transports/http/endpoints/api/users/endpoints.go

```diff
@@ -71,11 +69,16 @@ func (h *handler) register(c *gin.Context) {
 
 	// Check if user with this email already exists or there is another error
 	if err != nil {
-		if err == users.ErrUserAlreadyExists {
-			c.JSON(http.StatusBadRequest, "The email address is already associated with another user.")
-			return
+		switch err.(type) {
```

Comment:
It seems in all cases (beside `default`) we could use only one type of error, e.g. UserError with descriptive message from the throwing point (and as I can see you already add good decriptions for paymail and xpub errors).
Right now we will end up with messages like : `Error registering Paymail: error registering paymail in BUX: (internal error)`

Reply:
not really, this methods can throw 3 types of errors UserError PaymailError and XPubError as per the kind of api calls is doing .. and in switch u can see iam checking each type of the error instead comparing string